### DISCUSSION TITLE
ResteasyJackson2Provider without default modules

### DISCUSSION THREAD

**Discussion by unru006** [resteasy/resteasy#3268] [Ideas]
Hi resteasy-community :-)

We had/have an issue with the `ResteasyJackson2Provider `where in the readFrom and writeTo method default modules are going to be added:
```java
   @Override
   public Object readFrom(Class<Object> type, final Type genericType, Annotation[] annotations, MediaType mediaType, MultivaluedMap<String,String> httpHeaders, InputStream entityStream)
         throws IOException
   {
      LogMessages.LOGGER.debugf("Provider : %s,  Method : readFrom", getClass().getName());
      ClassAnnotationKey key = new ClassAnnotationKey(type, annotations);
      JsonEndpointConfig endpoint;
      endpoint = _readers.get(key);
      // not yet resolved (or not cached any more)? Resolve!
      if (endpoint == null) {
         ObjectMapper mapper = addDefaultModules(locateMapper(type, mediaType));
         ...
      }
   ...
   }

   private static ObjectMapper addDefaultModules(final ObjectMapper mapper) {
      return mapper
              // Register the mapping for Jakarta XML Binding Annotations for JSON Processing
              .registerModule(new JakartaXmlBindAnnotationModule());
   }
```
We are using a custom object mapper and register our required modules. Exactly this default module `JakartaXmlBindAnnotationModule`, we do not want to have in our object mapper.

I suggest that there should be kind of configuration to prevent the automatic inclusion of this module - is that possible and reasonable?

**Comment by jamezp** [comment]
It is possible and reasonable. Thank you for bringing this up as the same thing came up in Quarkus. I've filed https://issues.redhat.com/browse/RESTEASY-3223 for this. @unru006 do you have an example by chance of how you're creating the `ObjectMapper`?

**Reply by unru006** [reply]
Thanks for your quick reply, and sorry for my late one!
We are adding it with the help of the `jakarta.ws.rs.ext.Provider` annotation and extend from `jakarta.ws.rs.ext.ContextResolver<ObjectMapper>`:

```java
@Provider
@Produces(MediaType.APPLICATION_JSON)
public class StandaloneObjectMapper implements ContextResolver<ObjectMapper> {

    private final ObjectMapper objectMapper;

    public StandaloneObjectMapper() {
        ...
    }

    @Override
    public ObjectMapper getContext(final Class<?> type) {
        return objectMapper;
    }
}
```

`ResteasyJackson2Provider` finds it then, if I understood correctly.

Maybe a background information why we want to remove the `JakartaXmlBindAnnotationModule `module from registered modules: We use the objectMapper for Json-Mapping only but process POJOs which have mapping annotation that should target XML-Mapping only, like `jakarta.xml.bind.annotation.adapters.XmlJavaTypeAdapter`. Sadly this annotation is taken into account by JSON-Mapping as well. Thats why we do not want the related module registered.

**Reply by jamezp** [reply]
That makes sense to me. I went ahead and removed it https://github.com/resteasy/resteasy/pull/3280. It changes the defaults as users now need to add the module if they want to use it, but that seems reasonable to me. I don't think the `ObjectMapper.copy().addModule(...)` would work here because it would still use the annotations.

Given that and given the fact there is no way to remove the module for the user it seems better to require them to add it.
